src/clj/ataru/hakija/hakija_routes.clj

```diff
@@ -28,28 +29,36 @@
 
 (def ^:private cache-fingerprint (System/currentTimeMillis))
 
+(defn- deleted? [{:keys [deleted]}]
+  (and (some? deleted)
+       (t/after? (t/now) deleted)))
+
 (defn- fetch-form-by-key [key]
   (let [form (form-store/fetch-by-key key)]
-    (if form
+    (if (and (some? form)
+             (not (deleted? form)))
       (-> form
           (koodisto/populate-form-koodisto-fields)
           (response/ok))
       (response/not-found form))))
 
 (defn- handle-application [application]
   (info "Received application:" application)
-  (if (validator/valid-application? application)
-    (let [application-id        (application-store/add-new-application application)
-          person-service-job-id (job/start-job hakija-jobs/job-definitions
-                                               (:type person-integration/job-definition)
-                                               {:application-id application-id})]
-      (application-email/start-email-confirmation-job application-id)
-      (info "Stored application with id:" application-id)
-      (info "Started person creation job (to person service) with job id" person-service-job-id)
-      (response/ok {:id application-id}))
-    (do
-      (error "Invalid application!")
-      (response/bad-request))))
+  (let [form (form-store/fetch-latest-version (:form application))]
+    (if (and (some? form)
+             (not (deleted? form))
```

Comment:
Should we have some kind of error-message telling us that the form is no longer there (deleted). Otherwise it might be hard to debug these situations in production.